projects/plugins/beta/src/class-autoupdateself.php

```diff
@@ -234,18 +234,22 @@ public function get_plugin_info( $result, $action, $args ) {
 	 * The download from GitHub will produce a directory named like "Automattic-jetpack-beta-xxxxxxx".
 	 * We need to correct that so it will overwrite the current instance of the plugin.
 	 *
-	 * @param string $source File source location. Something like "/path/to/workdir/Automattic-jetpack-beta-xxxxxxx/".
-	 * @param string $remote_source Remote file source location. Something like "/path/to/workdir/".
-	 * @return string $source
+	 * @param string|WP_Error $source File source location. Something like "/path/to/workdir/Automattic-jetpack-beta-xxxxxxx/".
+	 * @param string          $remote_source Remote file source location. Something like "/path/to/workdir/".
+	 * @return string|WP_Error $source
 	 */
 	public function upgrader_source_selection( $source, $remote_source ) {
 		global $wp_filesystem;
-		if ( strstr( $source, '/Automattic-jetpack-beta-' ) ) {
+		if ( is_string( $source ) && strstr( $source, '/Automattic-jetpack-beta-' ) ) {
```

Comment:
```suggestion
		if ( is_string( $source ) && str_contains( $source, '/Automattic-jetpack-beta-' ) ) {
```

Reply:
I'll have to double check that the polyfill is loaded in the code path here. I think it probably is but I'm not 100% sure.

Reply:
55e2b38dc9